pkg/gw/safezadd_handler_test.go

```diff
@@ -0,0 +1,159 @@
+package gw
+
+import (
+	"encoding/base64"
+	"encoding/json"
+	"net/http"
+	"net/http/httptest"
+	"os"
+	"strconv"
+	"strings"
+	"testing"
+	"time"
+
+	immuclient "github.com/codenotary/immudb/pkg/client"
+	immudb "github.com/codenotary/immudb/pkg/server"
+	"github.com/grpc-ecosystem/grpc-gateway/runtime"
+)
+
+func TestSafeZAdd(t *testing.T) {
+	setName := base64.StdEncoding.EncodeToString([]byte("Soprano"))
+	uknownKey := base64.StdEncoding.EncodeToString([]byte("Marias Callas"))
+	tcpPort := generateRandomTCPPort()
+	//MetricsServer must not be started as during tests because prometheus lib panis with: duplicate metrics collector registration attempted
+	op := immudb.DefaultOptions().WithPort(tcpPort).WithDir("db_" + strconv.FormatInt(int64(tcpPort), 10)).WithMetricsServer(false)
+	s := immudb.DefaultServer().WithOptions(op)
+	go s.Start()
+	time.Sleep(2 * time.Second)
+	defer func() {
+		s.Stop()
+		time.Sleep(2 * time.Second) //without the delay the db dir is deleted before all the data has been flushed to disk and results in crash.
```

Comment:
In order to slower down tests execution with sleeps we could add a method on immudb like store.isClosed or similar and then loop a that bool. What's your opinion?

Reply:
> we could add a method on immudb like store.isClosed or similar and then loop a that bool.

It's not worth it just for tests. At a later point I'd rather refactor these tests to run in parallel and two seconds for each in parallel won't add too much. I will open an issue for this so I won't forget it.
I'd say let's leave it as it is for the moment as there are not many tests.

Reply:
Ok perfect